File: 448-find-all-numbers-disappeared-in-an-array/find-all-numbers-disappeared-in-an-array.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        int n = nums.length;
        List<Integer> numbers = new ArrayList<>();
        boolean[] present = new boolean[n + 1];

        for (int i = 0; i < n; i++) {
            present[nums[i]] = true;
        } 

        for (int i = 1; i <= n; i++) {
            if (!present[i]) {
                numbers.add(i);
            }
        }
        return numbers;
    }
}

File: 448-find-all-numbers-disappeared-in-an-array/find-all-numbers-disappeared-in-an-array.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        vector<int> result;
        vector<bool> seen(size(nums) + 1, false);
        for (auto c : nums)
            seen[c] = true;
        for (int i = 1; i <= size(nums); i++)
            if (!seen[i])
                result.push_back(i);
        return result;
    }
};